otherlibs/stdune-unstable/temp.ml

```diff
@@ -12,11 +14,11 @@ let try_paths n ~dir ~prefix ~suffix ~f =
       Path.relative dir (Printf.sprintf "%s%06x%s" prefix rnd suffix)
     in
     if n = 1 then
-      f path
+      try f path with
+      | Retry -> Code_error.raise "try_paths failed to find a good candidate" []
     else
-      match f path with
-      | exception _ -> loop (n - 1)
-      | r -> r
+      try f path with
+      | Retry -> loop (n - 1)
```

Comment:
It should be possible to switch the order of `if` and `try`:

```
match f path with
| exception Retry -> if n = 1 ...
| s -> ..
```

Reply:
Good idea! Done.